test/battle/move_effect/curse.c

```diff
@@ -34,3 +34,36 @@ SINGLE_BATTLE_TEST("Curse cuts the user's HP in half when used by Ghost-types")
         HP_BAR(player, hp: maxHP / 2);
     }
 }
+
+SINGLE_BATTLE_TEST("Curse applies to the user if used with Protean")
+{
+    GIVEN {
+        PLAYER(SPECIES_KECLEON) { Ability(ABILITY_PROTEAN); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_CURSE, target: player); }
+    } SCENE {
+        s32 playerMaxHP = GetMonData(&PLAYER_PARTY[0], MON_DATA_MAX_HP);
+        ABILITY_POPUP(player, ABILITY_PROTEAN);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_CURSE, player);
+        HP_BAR(player, damage: playerMaxHP / 2);
+        HP_BAR(player, damage: playerMaxHP / 4);
```

Comment:
why is the player damaged twice?

Reply:
End of turn Curse damage

Reply:
why is it inflicted to the player and not opponent?